core/corehttp/gateway_test.go

```diff
@@ -27,41 +26,47 @@ import (
 
 type mockNamesys map[string]path.Path
 
-func (m mockNamesys) Resolve(ctx context.Context, name string, opts ...nsopts.ResolveOpt) (value path.Path, err error) {
-	cfg := nsopts.DefaultResolveOpts()
+func (m mockNamesys) Resolve(ctx context.Context, p path.Path, opts ...namesys.ResolveOption) (namesys.ResolveResult, error) {
+	cfg := namesys.DefaultResolveOptions()
 	for _, o := range opts {
 		o(&cfg)
 	}
 	depth := cfg.Depth
-	if depth == nsopts.UnlimitedDepth {
+	if depth == namesys.UnlimitedDepth {
 		// max uint
 		depth = ^uint(0)
 	}
+	var (
+		value path.Path
+	)
+	name := path.SegmentsToString(p.Segments()[:2]...)
 	for strings.HasPrefix(name, "/ipns/") {
 		if depth == 0 {
-			return value, namesys.ErrResolveRecursion
+			return namesys.ResolveResult{Path: value}, namesys.ErrResolveRecursion
 		}
 		depth--
 
-		var ok bool
-		value, ok = m[name]
+		v, ok := m[name]
 		if !ok {
-			return nil, namesys.ErrResolveFailed
+			return namesys.ResolveResult{}, namesys.ErrResolveFailed
 		}
+		value = v
 		name = value.String()
 	}
-	return value, nil
+
+	value, err := path.Join(value, p.Segments()[2:]...)
+	return namesys.ResolveResult{Path: value}, err
 }
 
-func (m mockNamesys) ResolveAsync(ctx context.Context, name string, opts ...nsopts.ResolveOpt) <-chan namesys.Result {
-	out := make(chan namesys.Result, 1)
-	v, err := m.Resolve(ctx, name, opts...)
-	out <- namesys.Result{Path: v, Err: err}
+func (m mockNamesys) ResolveAsync(ctx context.Context, p path.Path, opts ...namesys.ResolveOption) <-chan namesys.ResolveAsyncResult {
+	out := make(chan namesys.ResolveAsyncResult, 1)
+	res, err := m.Resolve(ctx, p, opts...)
+	out <- namesys.ResolveAsyncResult{Path: res.Path, TTL: res.TTL, LastMod: res.LastMod, Err: err}
```

Comment:
:thought_balloon: `Result` looked way better, could we rename to that? (`ResolveAsyncResult` seems to be used the most, and it is all over the place, shorter name here would be better for readability)

Reply:
See conversation on Boxo side. I renamed to a simpler name.